### ISSUE TITLE
Switch build.sh to a Makefile

### ISSUE THREAD

**Issue by dlorenc** [kubernetes/minikube#40]
Once the localkube merge lands (#39 ), the build is getting complicated enough to deserve a real makefile. We'll want to building localkube for a different OS than minikube in most cases, with different sets of flags (static vs. dynamic linking, etc.).

**Comment by dlorenc** [comment]
cc @ethernetdan

**Comment by luxas** [comment]
+1 for this of course.
If it gets _even_ more complex, there's reason to move scripts (bash) to `build/` like k8s does.

**Comment by ethernetdan** [comment]
Since minikube's primary usecase for localkube is providing a Kube cluster within a Linux environment, I definitely think we should optimize for that usecase.

The current build step in the PR compiles it statically using an external linker while stripping symbols:
`go build --ldflags '-extldflags "-static" --s -w' k8s.io/minikube/cmd/localkube`

The static compilation was intended to support running from a "scratch" Docker image, which is not a requirement in minikube. The external linker usage is to support static compilation with cAdvisor.

If we decide to support other platforms we should probably use conditional compilation to remove the components that require Linux features (ie. proxy, kubelet) when targeting platforms other than Linux. This would require having changing components to register themselves in init blocks.

**Comment by luxas** [comment]
`localkube` targets: `linux/amd64`, `linux/arm`, `linux/arm64`, `linux/ppc64le`. Dynamically built until we fix cAdvisor: https://github.com/google/cadvisor/pull/1237 and `kubelet` in turn. I've already done the hard parts with dynamic cross-compilation (which is a bit tricky) with Kubernetes, so we'll just reuse that code.

`minikube` targets: `linux/amd64`, `linux/arm`, `linux/arm64`, `linux/ppc64le`, `darwin/amd64`, `windows/amd64`. Statically built. Should offer both a VM, dockerized and maybe host env for running `localkube`.

**Comment by dlorenc** [comment]
Resolved by #43
